### PR TITLE
First pass support for Melk's store, quick 'n dirty dropdown

### LINKED ISSUE #6: melks inventory

### PR DESCRIPTION
**Pull request by Coconutcoo**
Fixes #6 

Unfortunately starting running into a real large refactor if I tried to make char and storeType required arguments, since we've got a few other things like `Array.find<T>()` and `useSWR<T>()` that return `T|undefined`, so I've settled by making their types specifically `Character|undefined` etc.

The dropdown selector for store type is really a quick fix, I have no doubt a better UI can and should be done, but at least it gets the data accessible for now.

### CODE REVIEW

File: src/components/Store.tsx [hunk 1]
```diff
@@ -145,8 +146,9 @@ let filterOptions = {
 export type FilterOption = keyof typeof filterOptions
 export const FILTER_OPTIONS = Object.keys(filterOptions) as FilterOption[]
 
-export function Store({ character, sortOption, filterOption }: { character?: Character, sortOption: SortOption, filterOption: FilterOption }) {
-	let store = useStore(character)
+export function Store({ character, storeType, sortOption, filterOption }: { character: Character|undefined, storeType: StoreType, sortOption: SortOption, filterOption: FilterOption }) {
```
Comment (Chris-Bitler) [inline src/components/Store.tsx]: nitpick: For readability purposes, there should probably be a space on either side of the pipe symbol between Character and undefined
Also, should the log below this line be removed?

File: src/hooks/useStore.ts [hunk 2]
```diff
@@ -1,16 +1,16 @@
 import useSWR from "swr"
 import { useFetcher } from "./useFetcher"
-import type { Character, Store } from "../types"
+import type { Character, Store, StoreType } from "../types"
 import { useEffect } from "react"
 
-let key = (char?: Character) =>
+let key = (char: Character|undefined, storeType: StoreType) =>
```
Comment (Chris-Bitler) [inline src/hooks/useStore.ts]: same as above re: the spacing around the pipe symbol, here and below

**Review by Chris-Bitler — COMMENTED** [review]
Just a few small code style nitpicks

File: src/components/Store.tsx (hunk 1 above)
Comment (Coconutcoo) [reply src/components/Store.tsx]: Ah, good spot on the log line. Have added spaces to bring it into line with the rest of the code too.

File: src/hooks/useStore.ts [hunk 3]
```diff
@@ -1,16 +1,16 @@
 import useSWR from "swr"
 import { useFetcher } from "./useFetcher"
-import type { Character, Store } from "../types"
+import type { Character, Store, StoreType } from "../types"
 import { useEffect } from "react"
 
-let key = (char?: Character) =>
+let key = (char: Character | undefined, storeType: StoreType) =>
```
Comment (danreeves) [inline src/hooks/useStore.ts:6]: Is there a reason for changing `char?: Character` to `char: Charatcer | undefined`. AFAIK they are pretty much the same (optional params will take undefined as a valid value)

File: src/types.ts [hunk 4]
```diff
@@ -200,3 +200,5 @@ interface PlayerItems {
 	href: string
 	version: string
 }
+
+export type StoreType = 'credits' | 'marks'
```
Comment (danreeves) [inline src/types.ts]: Suggestion:

```ts
export const STORE_TYPES = ['credits', 'marks'] as const
export type StoreType = typeof STORE_TYPES[number]
```

That way when you're constructing the `<select>` you can map STORE_TYPES to `<option>`s and the options & type will always be in sync

**Review by danreeves — COMMENTED** [review]
Looks good. Just a couple of comments, but we could probably merge as is and clean up later

File: src/hooks/useStore.ts (hunk 3 above)
Comment (Coconutcoo) [reply src/hooks/useStore.ts:6]: A required parameter can't follow an optional one, so it was either explicitly say char can be undefined, or also make storeType optional (which doesn't make any sense - it's got a default value of credits and can't ever be undefined).

I felt this was the nicer way of the two, but am open to changing it. As you say, they're literally equivalent, but T|undefined has the advantage of a more consistent call signature.

File: src/types.ts (hunk 4 above)
Comment (Coconutcoo) [reply src/types.ts]: Great idea, will implement shortly

**Comment by Coconutcoo** [comment]
@danreeves I think all the feedback has been solved - I also removed the 3 `{" "}` after each sort, since it didn't seem to visually change the page. I assume you were trying to space out the options? It was entirely removed in the actual injected html, so I think it was doing nothing.

Please lmk if there was some deeper use for them in React that I've missed and I can re-add.

**Review by danreeves — APPROVED** [review]

**Comment by danreeves** [comment]
> the 3 {" "} after each sort

Yeah those look unnecessary

**Comment by danreeves** [comment]
Just tested. works well!

**Pull request merged**